File: CPP/HotSummer.cpp
#include <cstdio>

const int TIME = 9;

int main () {
	int cases;
	scanf("%d", &cases);
	while (cases--) {
		int targetedPowerUsage;
		scanf("%d", &targetedPowerUsage);
		for (int i = 0; i < TIME; i++) {
			int usage; scanf("%d", &usage);
			targetedPowerUsage -= usage;
		}
		if (targetedPowerUsage >= 0) fputs("YES\n", stdout);
		else fputs("NO\n", stdout);
	}
}

File: JAVA/HotSummer.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;


public class HotSummer {

	public static void main(String[] args) {
		try {
			BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
			int cases = Integer.parseInt(reader.readLine());
			
			while (cases-- > 0) {
				int targetedPowerUsage = Integer.parseInt(reader.readLine());
				String[] powerUsage = reader.readLine().split(" ");
				for (int i = 0; i < powerUsage.length; i++) targetedPowerUsage -= Integer.parseInt(powerUsage[i]);
				if (targetedPowerUsage >= 0) writer.append("YES\n");
				else writer.append("NO\n");
			}
			writer.flush();
			writer.close();
			reader.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

}
